sale_global_discount/models/sale_order.py

```diff
@@ -0,0 +1,140 @@
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+    amount_untaxed_before_global_discounts = fields.Monetary(
+        string='Amount Untaxed Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+    amount_total_before_global_discounts = fields.Monetary(
+        string='Amount Total Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+
+    @api.model
+    def get_discounted_global(self, price=0, discounts=None):
+        """Compute discounts successively"""
+        discounts = discounts or []
+        if not discounts:
+            return price
+        discount = discounts.pop(0)
+        price *= 1 - (discount / 100)
+        return self.get_discounted_global(price, discounts)
+
+    @api.depends('order_line.price_total', 'global_discount_ids')
+    def _amount_all(self):
+        res = super()._amount_all()
+        for order in self:
+            amount_untaxed_before_global_discounts = order.amount_untaxed
+            amount_total_before_global_discounts = order.amount_total
+            discounts = order.global_discount_ids.mapped('discount')
+            amount_discounted_untaxed = amount_discounted_tax = 0
+            for line in order.order_line:
+                discounted_subtotal = self.get_discounted_global(
+                    line.price_subtotal, discounts.copy())
+                amount_discounted_untaxed += discounted_subtotal
+                discounted_tax = line.tax_id.compute_all(
+                    discounted_subtotal, line.order_id.currency_id,
+                    1.0, product=line.product_id,
+                    partner=line.order_id.partner_shipping_id)
+                amount_discounted_tax += sum(
+                    t.get('amount', 0.0)
+                    for t in discounted_tax.get('taxes', []))
+            order.update({
+                'amount_untaxed_before_global_discounts': (
+                    amount_untaxed_before_global_discounts),
+                'amount_total_before_global_discounts': (
+                    amount_total_before_global_discounts),
+                'amount_global_discount': (
+                    amount_untaxed_before_global_discounts -
+                    amount_discounted_untaxed),
+                'amount_untaxed': amount_discounted_untaxed,
+                'amount_tax': amount_discounted_tax,
+                'amount_total': (
+                    amount_discounted_untaxed + amount_discounted_tax),
+            })
+        return res
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        res = super().onchange_partner_id()
+        if self.partner_id.customer_global_discount_ids:
+            self.global_discount_ids = (
+                self.partner_id.customer_global_discount_ids or
+                self.parnter_id.commercial_partner_id
```

Comment:
```suggestion
                self.partner_id.commercial_partner_id
```

Can you check if this bug is also in the original version?

Reply:
Yes, I have suggested the module author to correct it.